### ISSUE TITLE
Rename doesn't report conflict from overflow triggered by nameof

### ISSUE THREAD

**Issue by SLaks** [dotnet/roslyn#6304]
_astoudningly evil corner case_

```C#
const int a = (nameof(a) == "a" ? 1 : 2) * int.MaxValue;
```

Renaming `a` to any other name will induce `CS0220	The operation overflows at compile time in checked mode`.

Ideally, rename should catch such compiler errors and report an unresolvable conflict.
This would be especially important if the conflict occurs in another file and the project already has other errors (so that the error in the error list would be less noticeable)

**Comment by CyrusNajmabadi** [comment]
I don't seem why this is conflict.  This seems like the bog standard rename case of "the code actually depends on the name of something, and thus changes behavior when you rename something".  This is already trivial to run into with many cases (esp. around reflection).  

I don't believe we ever claimed that rename wouldn't change any compile time or runtime behavior.  After all, if you have an app that just prints out the names of symbols, it will not break when you rename something.  All we state is that we rename references to the symbol, and we will do a best effort to prevent code from changing which symbols it binds to because of the rename.

**Comment by CyrusNajmabadi** [comment]
Closing as by design.
